### PR TITLE
[Car]: add controller, service, repository

### PR DESCRIPTION
**Pull request by Someboty**
Create CarController with add(), getAll(), getById(), update(), delete() methods
Create CarService and CarServiceImpl with corresponding methods 
Create CarDto and CreateCarDto for data protection 
Create MapperConfig for mapper classes configuration 
Create CarMapper for mapping between CreateCarDto, CarDto and Car objects 
Create CarRepository for DB operations
Create EntityNotFoundException for exceptions related to the absence of corresponding objects in the database

### CODE REVIEW

File: src/main/java/com/team01/carsharingapp/service/impl/CarServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        car.setAmountAvailable(request.amountAvailable());
+        car.setDailyFee(request.dailyFee());
+        return carMapper.toDto(carRepository.save(car));
+    }
+
+    @Override
+    @Transactional
+    public void delete(Long id) {
+        if (!carRepository.existsById(id)) {
+            throw new EntityNotFoundException("Can't find book by id: " + id);
```
Comment (ConfusedSatlan) [inline src/main/java/com/team01/carsharingapp/service/impl/CarServiceImpl.java:59]: change book to car

**Review by ConfusedSatlan — CHANGES_REQUESTED** [review]
Let's improve your solution

File: src/main/java/com/team01/carsharingapp/service/impl/CarServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    @Override
+    @Transactional
+    public CarDto update(Long id, CreateCarRequestDto request) {
+        Car car = carById(id);
+        car.setModel(request.model());
+        car.setBrand(request.brand());
+        car.setType(Car.Type.valueOf(request.type()));
+        car.setAmountAvailable(request.amountAvailable());
+        car.setDailyFee(request.dailyFee());
```
[the remark this answers is not in the document]
Comment (Someboty) [reply src/main/java/com/team01/carsharingapp/service/impl/CarServiceImpl.java:51]: It's not for creating, it's for updating. But I move it to separate method, as you wish.

File: src/main/java/com/team01/carsharingapp/service/impl/CarServiceImpl.java (hunk 1 above)
Comment (Someboty) [reply src/main/java/com/team01/carsharingapp/service/impl/CarServiceImpl.java:59]: My bad

**Review by IhorKalaur — APPROVED** [review]

**Review by keda-vlad — APPROVED** [review]

File: src/main/java/com/team01/carsharingapp/service/impl/CarServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        carRepository.deleteById(id);
+    }
+
+    private Car carById(Long id) {
+        return carRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find car by id: " + id)
+        );
+    }
+
+    private void updateValues(Car car, CreateCarRequestDto request) {
```
[the remark this answers is not in the document]
Comment (Someboty) [reply src/main/java/com/team01/carsharingapp/service/impl/CarServiceImpl.java:66]: Cannot resolve symbol 'builder'

**Review by ConfusedSatlan — APPROVED** [review]

**Pull request merged**